### PR TITLE
feat(PartnerNetwork) : added bpnIds field to search active company BPN ids

### PR DESCRIPTION
**Pull request by VPrasannaK94**
## Description

added bpnIds field to search active company BPN ids

## Why

added bpnIds field to search active company BPN ids

## Issue

N/A Ref : Ref : CPLP-3109

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/administration/Administration.Service/Controllers/PartnerNetworkController.cs [hunk 1]
```diff
@@ -56,6 +57,6 @@ public PartnerNetworkController(IPartnerNetworkBusinessLogic logic)
     [Route("memberCompanies")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [PublicUrl(CompanyRoleId.ACTIVE_PARTICIPANT, CompanyRoleId.SERVICE_PROVIDER, CompanyRoleId.APP_PROVIDER)]
-    public IAsyncEnumerable<string?> GetAllMemberCompaniesBPNAsync() =>
-        _logic.GetAllMemberCompaniesBPNAsync();
+    public IAsyncEnumerable<string?> GetAllMemberCompaniesBPNAsync([FromBody] IEnumerable<string> bpnIds) =>
```
Comment (Phil91) [inline src/administration/Administration.Service/Controllers/PartnerNetworkController.cs:60]: Please make sure that the bpnIds are optional. Mentioned in the ticket, without submitting any bpn the endpoint should still return all active companies.

I'd suggest renaming the bpnIds to bpns as well.

File: tests/portalbackend/PortalBackend.DBAccess.Tests/CompanyRepositoryTests.cs [hunk 2]
```diff
@@ -902,6 +902,30 @@ public async Task CheckBpnExists_WithValid_ReturnsTrue()
 
     #endregion
 
+    #region GetAllMemberCompaniesBPN
+
+    [Fact]
+    public async Task GetAllMemberCompaniesBPN()
```
Comment (Phil91) [inline tests/portalbackend/PortalBackend.DBAccess.Tests/CompanyRepositoryTests.cs:908]: Please add a unit test for an empty or null set of bpns

**Review by Phil91 — CHANGES_REQUESTED** [review]

File: src/administration/Administration.Service/Controllers/PartnerNetworkController.cs [hunk 3]
```diff
@@ -56,6 +57,6 @@ public PartnerNetworkController(IPartnerNetworkBusinessLogic logic)
     [Route("memberCompanies")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [PublicUrl(CompanyRoleId.ACTIVE_PARTICIPANT, CompanyRoleId.SERVICE_PROVIDER, CompanyRoleId.APP_PROVIDER)]
-    public IAsyncEnumerable<string?> GetAllMemberCompaniesBPNAsync() =>
-        _logic.GetAllMemberCompaniesBPNAsync();
+    public IAsyncEnumerable<string?> GetAllMemberCompaniesBPNAsync([FromBody] IEnumerable<string>? bpnIds) =>
```
Comment (Phil91) [inline src/administration/Administration.Service/Controllers/PartnerNetworkController.cs:60]: FromBody isn't possible for Get requests. Please change to FromQuery

```suggestion
    public IAsyncEnumerable<string?> GetAllMemberCompaniesBPNAsync([FromQuery] IEnumerable<string>? bpnIds) =>
```

**Review by Phil91 — CHANGES_REQUESTED** [review]
One minor thing, besides of that everything seems fine.
Please test the changes locally with postman the next time

**Review by Phil91 — APPROVED** [review]
